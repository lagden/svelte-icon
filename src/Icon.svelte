<script>
	export let name
	export let useBase = true
	export let useTransition = false
</script>

<svg
	class:_tadashi_svelte_icon___base={useBase}
	class:_tadashi_svelte_icon___transition={useTransition}
	on:click
	{...$$restProps}
><use xlink:href="#{name}" /></svg>

<style>
	._tadashi_svelte_icon___base {
		fill: var(--tadashi_svelte_icon_fill, initial);
		width: var(--tadashi_svelte_icon_width, initial);
		height: var(--tadashi_svelte_icon_height, initial);
	}

	._tadashi_svelte_icon___transition {
		transition: var(--tadashi_svelte_icon_transition, fill .4s);
	}
</style>
