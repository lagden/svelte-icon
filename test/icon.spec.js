/* globals describe, afterEach, test, expect */

import {cleanup, render} from '@testing-library/svelte'
import Icon from '../src/icon.js'

describe('Icon', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(Icon, {
			props: {
				name: 'icon_name',
			},
		})
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Icon, {
			props: {
				style: '--tadashi_svelte_icon_fill: green;--tadashi_svelte_icon_width: 30px',
				name: 'icon_name',
			},
		})
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Icon, {
			props: {
				name: 'icon_name',
				useBase: false,
				useTransition: true,
			},
		})
		expect(container).toMatchSnapshot()
	})
})
