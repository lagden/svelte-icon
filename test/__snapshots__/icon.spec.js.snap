// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match snapshot default 1`] = `
<body>
  <div>
    <svg
      class="_tadashi_svelte_icon___base svelte-1g6tlhj"
    >
      <use
        xlink:href="#icon_name"
      />
    </svg>
  </div>
</body>
`;

exports[`Icon should match snapshot props 1`] = `
<body>
  <div>
    <svg
      class="_tadashi_svelte_icon___base svelte-1g6tlhj"
      style="--tadashi_svelte_icon_fill: green;--tadashi_svelte_icon_width: 30px"
    >
      <use
        xlink:href="#icon_name"
      />
    </svg>
  </div>
</body>
`;

exports[`Icon should match snapshot props 2`] = `
<body>
  <div>
    <svg
      class="_tadashi_svelte_icon___transition svelte-1g6tlhj"
    >
      <use
        xlink:href="#icon_name"
      />
    </svg>
  </div>
</body>
`;
